communication/communications.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+from serial import Serial
+from Queue import Queue
+import fileinput
+import readline
+from time import sleep
+from threading import Thread
+
+# constants
+TEAM = 0
+
+# Operation Codes (OpCodes)
+STOP = 0
+FORWARD = 1
+BACKWARD = 2
+LEFT = 3
+RIGHT = 4
+GRAB = 5
+
+
+
+class CommsToArduino(object):
+
+    queue = Queue()
+    internal_queue = Queue()
+    write_queue = Queue()
+    # these should be hard-coded, the values should not change
+    def __init__(self, port="/dev/ttyACM0",
+                 rate=115200,
+                 timeout=0,
+                 connected=False):
+        self.isConnected = connected
+        self.port = port
+        self.comn = None # updated when we establish the connection
+        self.rate = rate
+        self.timeout = timeout
+        self.connect()
+
+    seqNo = False
+    ready = True
+
+    # this function establishes the connection between the devices
+    # and updates the boolean variable isConnected
+    def connect(self):
+        if self.isConnected is False and self.comn is None:
+            try:
+                self.comn = Serial(port=self.port,
+                                   baudrate=self.rate,
+                                   timeout=self.timeout)
+                self.isConnected = True
+            except OSError as ex:
+                print("Cannot connect to Arduino.")
+                print(ex.strerror)
+
+    def create_checksum(self, arg, opcode):
+        """
+        Creates the checksum that is used to verify the opcode.
+        """
+
+        return (arg + opcode) % 10
+
+    def _write(self, sig, opcode, arg, seqNo):
+        """
+        Sends the code to the Arduino
+        """
+        checksum = self.create_checksum(arg, opcode)
+        opcode_string = "%d%d%03d%d%d\r" % (sig, opcode, arg, checksum, seqNo)
+        self.to_robot(opcode_string)
+        return
+
+    def to_robot(self, message):
```

Comment:
We should validate robot output (e.g. to see if checksum failed).

If checksum fails, use '2' instead of sequence code. (note: this code currently ignores checksums)